### PR TITLE
Group and tactical enhancements

### PR DESCRIPTION
**Pull request by nk3nny**
### Fixes errors and adds enhancements to group and individual tactical actions
Adds more robust stance changes and forceMove to tacticsFlank to induce more movement
Adds reset for 'ATTACK'-locked units in findReadyUnits  (fixes reporting units which are not ready)
Fixes doGroupFlanks tendency to not do anything!
Fixes doGroupAssault tendency to cause AI to look in circles!
Fixes doAssault building sorting distance to more reliably assault buildings

### CODE REVIEW

**Comment by nk3nny** [comment]
Not going to fix linter error until relevant for 2.5.1 release.

File: addons/main/functions/GroupAction/fnc_doGroupFlank.sqf [hunk 1]
```diff
@@ -21,23 +21,24 @@
...
-_units = _units select { _x call FUNC(isAlive) && { _x distance2D (_pos select 0) > 10 } && { !isPlayer _x } };
+_units = _units select { _x call FUNC(isAlive) && { _x distance2D _overwatch > 12 } && { !isPlayer _x } };
 _vehicles = _vehicles select { canFire _x };
 
 {
     private _posASL = AGLtoASL (selectRandom _pos);
 
     // stance
-    _x setUnitPosWeak "MIDDLE";
+    _x setUnitPos (["MIDDLE", "DOWN"] select (getSuppression _unit > 0.33));
```
Comment (jokoho48) [inline addons/main/functions/GroupAction/fnc_doGroupFlank.sqf:31]: i think this needs to be _x and not _unit
```suggestion
    _x setUnitPos (["MIDDLE", "DOWN"] select (getSuppression _x > 0.33));
```

File: addons/danger/functions/fnc_tacticsSuppress.sqf [hunk 2]
```diff
@@ -22,6 +22,13 @@ params ["_unit", "_target", ["_units", []], ["_delay", 17]];
 // find target
 _target = _target call CBA_fnc_getPos;
 
+// exit with flank if visible distance is far actual target (within one grid : 71 * 71 meters)
+private _targetASL = ATLtoASL _target;
+private _vis = lineIntersectsSurfaces [(eyePos _unit) vectorAdd [0, 0, 12], _targetASL vectorAdd [0, 0, 2.5], _unit, objNull, true, 1, "FIRE", "GEOM"];
+if (_vis isNotEqualTo [] && {((_vis select 0) select 0) vectorDistanceSqr _targetASL > 5041}) exitWith {
```
Comment (jokoho48) [inline addons/danger/functions/fnc_tacticsSuppress.sqf:28]: didn't we want to filter this a bit more?
so using `_vis findIf {(_x select 0) isKindOf "Building"};` is what we wanted to do I think

**Review by jokoho48 — APPROVED** [review]
just that little thing we need to discuss but else this looks good for now

File: addons/danger/functions/fnc_tacticsSuppress.sqf (hunk 2 above)
Comment (nk3nny) [reply addons/danger/functions/fnc_tacticsSuppress.sqf:28]: Will investigate now.

Comment (nk3nny) [reply addons/danger/functions/fnc_tacticsSuppress.sqf]: Added

**Review by diwako — APPROVED** [review]

**Review by jokoho48 — APPROVED** [review]

**Pull request merged**